logging/loggerConfig.go

```diff
@@ -0,0 +1,65 @@
+package logging
+
+import (
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+)
+
+type loggerConfig struct {
+	SchedTraceConfig struct {
+		Enabled           bool   `yaml:"enabled"`
+		FilenameExtension string `yaml:"filenameExtension"`
+		AllowOnConsole    bool   `yaml:"allowOnConsole"`
+	} `yaml:"schedTrace"`
+
+	PCPConfig struct {
+		Enabled           bool   `yaml:"enabled"`
+		FilenameExtension string `yaml:"filenameExtension"`
+		AllowOnConsole    bool   `yaml:"allowOnConsole"`
+	} `yaml:"pcp"`
+
+	ConsoleConfig struct {
+		Enabled           bool   `yaml:"enabled"`
+		FilenameExtension string `yaml:"filenameExtension"`
+		MinLogLevel       string `yaml:"minLogLevel"`
+		AllowOnConsole    bool   `yaml:"allowOnConsole"`
+	} `yaml:"console"`
+
+	SPSConfig struct {
+		Enabled           bool   `yaml:"enabled"`
+		FilenameExtension string `yaml:"filenameExtension"`
+		AllowOnConsole    bool   `yaml:"allowOnConsole"`
+	} `yaml:"sps"`
+
+	TaskDistrConfig struct {
+		Enabled           bool   `yaml:"enabled"`
+		FilenameExtension string `yaml:"filenameExtension"`
+		AllowOnConsole    bool   `yaml:"allowOnConsole"`
+	} `yaml:"clsfnTaskDistrOverhead"`
+
+	SchedWindowConfig struct {
+		Enabled           bool   `yaml:"enabled"`
+		FilenameExtension string `yaml:"filenameExtension"`
+		AllowOnConsole    bool   `yaml:"allowOnConsole"`
+	} `yaml:"schedWindow"`
+
+	Format []string `yaml:"format"`
+}
+
+func GetConfig(logConfigFilename string) (*loggerConfig, error) {
```

Comment:
does this need to be exported outside the package?

Reply:
No, not needed. Changed it to `getConfig` .